lara-typescript/src/interactive-api-host/pub-sub-manager.ts

```diff
@@ -0,0 +1,178 @@
+    let channel = this.channels.get(channelId);
+
+    if (!channel) {
+      channel = {
+        subscribers: new Set(),
+        publisherId: interactiveId,
+        channelInfo
+      };
+      this.channels.set(channelId, channel);
+    } else {
+      channel.publisherId = interactiveId;
+      channel.channelInfo = channelInfo;
+    }
+
+    // Send channelInfo to all current subscribers
+    const channelInfoMessage: IPubSubChannelInfo = {
+      channelId,
+      channelInfo,
+      publisherId: interactiveId,
+      timestamp: Date.now()
+    };
+
+    channel.subscribers.forEach((subscriber: SubscriberInfo) => {
+      const phone = this.phones.get(subscriber.interactiveId);
+      if (phone) {
+        phone.post("pubSubChannelInfo", channelInfoMessage);
+      }
+    });
+  }
+
+  /**
+   * Subscribe an interactive to a channel.
+   * If a publisher already exists for this channel, immediately send channelInfo to the new subscriber.
+   */
+  public subscribe(interactiveId: string, channelId: string, subscriptionId: string): void {
+    let channel = this.channels.get(channelId);
+
+    if (!channel) {
+      channel = {
+        subscribers: new Set()
+      };
+      this.channels.set(channelId, channel);
+    }
+
+    channel.subscribers.add({ interactiveId, subscriptionId });
+
+    // If a publisher already exists for this channel, send channelInfo to the new subscriber
+    if (channel.publisherId && channel.channelInfo !== undefined) {
+      const channelInfoMessage: IPubSubChannelInfo = {
+        channelId,
+        channelInfo: channel.channelInfo,
+        publisherId: channel.publisherId,
+        timestamp: Date.now()
+      };
+
+      const phone = this.phones.get(interactiveId);
+      if (phone) {
+        phone.post("pubSubChannelInfo", channelInfoMessage);
+      }
+    }
+  }
+
+  /**
+   * Unsubscribe an interactive from a channel.
+   */
+  public unsubscribe(interactiveId: string, channelId: string, subscriptionId: string): void {
+    const channel = this.channels.get(channelId);
+
+    if (!channel) {
+      return;
+    }
+
+    // Find and remove the subscriber
+    channel.subscribers.forEach((subscriber: SubscriberInfo) => {
+      if (subscriber.interactiveId === interactiveId && subscriber.subscriptionId === subscriptionId) {
+        channel.subscribers.delete(subscriber);
+      }
+    });
+
+    // Clean up empty channels
+    if (channel.subscribers.size === 0 && !channel.publisherId) {
+      this.channels.delete(channelId);
+    }
+  }
+
+  /**
+   * Publish a message to all subscribers on a channel.
+   */
+  public publish(publisherId: string, channelId: string, message: any): void {
+    const channel = this.channels.get(channelId);
+
+    if (!channel) {
+      return;
+    }
+
+    channel.lastPublishTime = Date.now();
+
+    const pubSubMessage: IPubSubMessage = {
+      channelId,
+      message,
+      publisherId,
```

Comment:
So this publisherId is different than the publisherId on the channel info. This is publisherId is the interactive that actually published the message while the other one is the interactive that last created the channel.

Are you using the info about which interactive created the channel somewhere already? Perhaps you could change that to "creatorId" instead.